chainerx_cc/examples/mnist_cc/train_mnist.cc

```diff
@@ -0,0 +1,205 @@
+
+    int64_t n_train = train_x.shape().front();
+    int64_t n_test = test_x.shape().front();
+    chainerx::Array train_indices = chainerx::Arange(n_train, chainerx::Dtype::kInt64);
+
+    // Initialize the model with random parameters.
+    Model model{train_x.shape()[1], n_hidden, 10, n_layers};
+    std::random_device rd{};
+    std::mt19937 gen{rd()};
+    std::normal_distribution<float> dist{0.f, 0.05f};
+    model.Initialize(gen, dist);
+
+    auto start = std::chrono::high_resolution_clock::now();
+
+    for (int64_t epoch = 0; epoch < epochs; ++epoch) {
+        // The underlying data of the indices' Array is contiguous so we can use std::shuffle to randomize the order.
+        assert(train_indices.IsContiguous());
+        std::shuffle(
+                reinterpret_cast<int64_t*>(train_indices.raw_data()), reinterpret_cast<int64_t*>(train_indices.raw_data()) + n_train, gen);
+        for (int64_t i = 0; i < n_train; i += batch_size) {
+            chainerx::Array indices = train_indices.At({chainerx::Slice{i, i + batch_size}});
+            chainerx::Array x = train_x.Take(indices, 0);
+            chainerx::Array t = train_t.Take(indices, 0);
+
+            chainerx::Backward(SoftmaxCrossEntropy(model(x), t));
+
+            // Vanilla SGD.
+            for (const chainerx::Array& p : model.params()) {
+                chainerx::Array p_as_grad_stopped = p.AsGradStopped();
+                p_as_grad_stopped -= p.GetGrad()->AsGradStopped() * lr;
+                p.ClearGrad();
+            }
+        }
+
+        // Evaluate.
+        {
+            chainerx::NoBackpropModeScope scope{};
+
+            chainerx::Array loss = chainerx::Zeros({}, chainerx::Dtype::kFloat32);
+            chainerx::Array acc = chainerx::Zeros({}, chainerx::Dtype::kFloat32);
+
+            for (int64_t i = 0; i < n_test; i += batch_size) {
+                std::vector<chainerx::ArrayIndex> indices{chainerx::Slice{i, i + batch_size}};
+                chainerx::Array x = test_x.At(indices);
+                chainerx::Array t = test_t.At(indices);
+                chainerx::Array y = model(x);
+                loss += SoftmaxCrossEntropy(y, t, false);
+                acc += (y.ArgMax(1).AsType(t.dtype()) == t).Sum().AsType(acc.dtype());
+            }
+
+            std::cout << "epoch: " << epoch << " loss=" << chainerx::AsScalar(loss / n_test)
+                      << " accuracy=" << chainerx::AsScalar(acc / n_test)
+                      << " elapsed_time=" << std::chrono::duration<double>{std::chrono::high_resolution_clock::now() - start}.count()
+                      << std::endl;
+        }
+    }
+}
+
+int main(int argc, char** argv) {
+    int64_t epochs{20};
+    int64_t batch_size{100};
+    int64_t n_hidden{1000};
+    int64_t n_layers{3};
+    float lr{0.01};
+    std::string device_name{"native"};
+    std::string mnist_root{"./"};
+
+    for (int i = 1; i < argc; i += 2) {
+        std::string arg = argv[i];
+        if (arg == "--epoch") {
```

Comment:
How about matching those in the existing examples?
e.g. `--n-hidden` -> `--unit`

Reply:
Fixed.